include/crow/http_response.h

```diff
@@ -233,7 +233,7 @@ namespace crow
 #ifdef CROW_ENABLE_COMPRESSION
             compressed = false;
 #endif
-            if (file_info.statResult == 0)
+            if (file_info.statResult == 0 && S_ISREG(file_info.statbuf.st_mode))
```

Comment:
This will most likely not work on Windows.

AFAIK Windows does not define `S_ISREG` so it needs to either be emulated or a windows specific check needs to be added.

[This StackOverflow question](https://stackoverflow.com/questions/11238918/s-isreg-macro-undefined) talks about the matter.

Reply:
Ok. No windows here, so I naturally forget this niche OS :) I'll try to add something that works on windows as well. Thanks :)

Reply:
C++17 introduces the filesystem API but for that two earlier standards would have to be dropped. I also don't believe having behavior depend upon the used standard is a particularly great idea...

Reply:
Agreed. Then there is boost filesystem. But I don't think introducing a new dependency just  for one function is a good idea either. Defining the macro for windows systems like it's done in the SO-discussion looks not too bad. I'd go for that. Just need to find a system I can test this on :D

Reply:
Well, we just dropped boost in its entirety as a dependency. 😅 I'll be happy to test it on my machine. 👍

Reply:
Questions:
1) is it ok to use `#if defined(_WIN32)` to test for windows builds or is `#if defined (_MSC_VER)` the preferred thing to do?
2) clang-format wants to reformat `(((m) & S_IFMT) == S_IFREG)` to `(((m)&S_IFMT) == S_IFREG)` (note the lack of spaces around `&`). Which doesn't look nice imo. What to do about this, if anything?

Reply:
1. I believe `#if defined (_MSC_VER)` is what Crow uses.
2. If it bothers you too much you can put `// clang-format off` before it and `// clang-format on` after. Frankly I don't think it's too big of a deal..